jira-macro/jira-macro-default/src/main/java/org/xwiki/contrib/jira/macro/JIRADataSource.java

```diff
@@ -41,5 +42,5 @@ public interface JIRADataSource
      * @throws MacroExecutionException in case of an error while getting the JIRA data
      */
     Collection<Element> getData(String macroContent, JIRAMacroParameters parameters)
-        throws MacroExecutionException;
+        throws MacroExecutionException, JIRABadRequestException;
```

Comment:
Do you really need it there? AFAICS that's the only place where you expose the new exception in an API and not only internally.

Reply:
Yes, because it could raise a `JIRABadRequestException` exception.